File: examples/golang/server/proto/msg_friend_base_add.go
package proto

import (
	"packet"
)

type MsgFriendBaseAdd struct {
	Uid                      uint32
	Uname                    string
}

func (this *MsgFriendBaseAdd) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.Uid)
	pack.WriteString(this.Uname)

	return pack.ReadBytes()
}

func MsgFriendBaseAddDecode(pack *packet.Packet) *MsgFriendBaseAdd {
	msgFriendBaseAdd := &MsgFriendBaseAdd{}

	msgFriendBaseAdd.Uid = pack.ReadUint32()
	msgFriendBaseAdd.Uname = pack.ReadString()
	return msgFriendBaseAdd
}

File: data/data.code.typescript.laya2.client/MsgFriendBaseAdd.ts
import Packet from './Packet'


export default class MsgFriendBaseAdd
{
	private _uid: number;
	private _uname: string;


	public Encode(): Packet {
		let packet: Packet = new Packet();
		packet.WriteUint(this._uid);
		packet.WriteString(this._uname);
		return packet;
	}


	constructor(packet?: Packet) {
		if (packet) {
			this._uid = packet.ReadUint();
			this._uname = packet.ReadString();
		}
	}

	public GetBuffer(): ByteBuffer
	{
		return this.Encode().GetBuffer();
	}


	public get uid(): number { return this._uid; }
	public set uid(value: number) { this._uid = value; }
	public get uname(): string { return this._uname; }
	public set uname(value: string) { this._uname = value; }
}
